File: app/src/test/java/com/socotech/swa/mvp/UnitTestSchedulerProvider.java
package com.socotech.swa.mvp;

import androidx.annotation.NonNull;
import io.reactivex.Scheduler;
import io.reactivex.schedulers.Schedulers;

import javax.inject.Singleton;

/**
 * Provides different types of schedulers.
 */
@Singleton
public class UnitTestSchedulerProvider implements SchedulerProvider {

    @NonNull
    @Override
    public Scheduler io() {
        return Schedulers.trampoline();
    }

    @NonNull
    @Override
    public Scheduler ui() {
        return Schedulers.trampoline();
    }

}

File: app/src/test/java/com/android/mvp/UnitTestSchedulerProvider.kt
package com.android.mvp

import io.reactivex.Scheduler
import io.reactivex.schedulers.Schedulers
import javax.inject.Singleton

/**
 * Provides different types of schedulers.
 */
@Singleton
class UnitTestSchedulerProvider : SchedulerProvider {

    override fun io(): Scheduler {
        return Schedulers.trampoline()
    }

    override fun ui(): Scheduler {
        return Schedulers.trampoline()
    }

}
